core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -311,6 +311,12 @@ final class ZSTM[-R, +E, +A] private[stm] (
       }
     )
 
+  /**
+   * "Peeks" at the success of transactional effect.
+   */
+  def tap[R1 <: R, E1 >: E](f: A => ZSTM[R1, E1, Any]): ZSTM[R1, E1, A] =
+    flatMap(f) *> self
```

Comment:
If this would be an effect it would be evalulated twice, right?
Is this OK since it’s STM?

Reply:
Good point. I think this should should be `flatMap(a => f(a).as(a))`.

Reply:
Yes, just started writing that one. One sec :)

Reply:
Done, good catch!